### ISSUE TITLE
optimize: Provide Completed Examples

### ISSUE THREAD

**Issue by rigelrozanski** [gonum/gonum#676]
Hey, I'd love to use this package, difficult for beginners without examples, mostly because of the many layers of custom types. It would be great to have a completed example for how to use this package for even a basic minimization problem... This repo does it in a few lines, would be great to have something similar (nonlinear optimization section is what I'm looking at) https://github.com/emef/go.ml

**Comment by btracey** [comment]
I agree more documentation is needed. There is this though which might help if you didn’t see it
https://godoc.org/gonum.org/v1/gonum/optimize#example-Minimize
